tests/scripts/ci_perf.py

```diff
@@ -0,0 +1,177 @@
+import os
+import shutil
+import subprocess
+import sys
+import glob
+from sys import version_info
+
+##########################################################################
+# Argument Parser
+##########################################################################
+
+description = 'Tool to simulate CI perf jobs'
+
+parser = argparse.ArgumentParser(description=description)
+
+#job arguments
+parser.add_argument('-job', dest='job', default=None, required=True, choices=['CoreCLR-Scenarios'])
+parser.add_argument('-isPR', dest='isPR', default=True, choices=[True, False])
+parser.add_argument('-arch', dest='arch', default='x64', choices=['x64', 'x86'])
+parser.add_argument('-os', dest='operatingSystem', default='Windows_NT', choices=['Windows_NT', 'Ubuntu16.04', 'Ubuntu14.04', 'OSX'])
+parser.add_argument('-optLevel', dest='optLevel', default='full_opt', choices=['full_opt', 'min_opt', 'tiered'])
+parser.add_argument('-jitName', dest='jitName', default='ryujit', choices=['ryujit'])
+parser.add_argument('-benchviewCommitName', dest='benchviewCommitName', default='FAKE_BV_COMMIT_NAME')
+parser.add_argument('-bvUploadSasToken', dest='bvUploadSasToken', default='FAKE_BV_SAS_UPLOAD_TOKEN')
+parser.add_argument('-uploadToBenchview', dest='uploadToBenchview', default=False, choices=[True, False])
+
+#ambient jenkins properties
+parser.add_argument('-workspace', dest='workspace', default=os.path.abspath(os.getcwd()))
+parser.add_argument('-gitBranchWithoutOrigin', dest='gitBranchWithoutOrigin', default='FAKE_BRANCHNAME')
+parser.add_argument('-gitBranch', dest='gitBranch', default='origin/FAKE_BRANCHNAME')
+parser.add_argument('-gitCommit', dest='gitCommit', default='FAKE_GIT_COMMIT')
+
+
+
+##########################################################################
+# Helper Functions
+##########################################################################
+
+def log(message):
+    """ Print logging information
+    Args:
+        message (str): message to be printed
+    """
+
+    print('[%s]: %s' % (sys.argv[0], message))
+
+def is_supported_version() -> bool:
+    """ Checkes that the version of python is at least 3.5
+    Returns:
+        bool : true if python version is 3.5 or higher
+    """
+    return version_info.major > 2 and version_info.minor > 4
+
+def run_command(runArgs, environment):
+    """ Run command specified by runArgs
+    Args:
+        runargs (str[]): list of arguments for subprocess
+        environment(str{}): dict of environment variable
+    """
+    log('RUNNING COMMAND: ' + runArgs)
+    subprocess.check_call(runArgs, env=environment)
+
+##########################################################################
+# CI jobs
+##########################################################################
+
+class jenkins_properties:
+    def __init__(self, workspace, git_branch_without_origin, git_branch, git_commit):
+        self.workspace = workspace
+        self.git_branch_without_origin = git_branch_without_origin
```

Comment:
As I mentioned above, all this information can be determined from the repo itself.

Reply:
As above, the local repo state may not accurately represent what the git state would be in CI. I'd rather have it be an obvious sentinel.